dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,42 +53,120 @@ public Pattern getParameterPattern() {
-            Map<String, DbParameterAccessor> allParams = new HashMap<String, DbParameterAccessor>();
-            int position = 0;
-            while (rs.next()) {
-                String columnName = rs.getString(1);
-                String dataType = rs.getString(2);
-                DbParameterAccessor dbp = createDbParameterAccessor(
-                        columnName,
-                        Direction.INPUT,
-                        typeMapper.getJDBCSQLTypeForDBType(dataType),
-                        getJavaClass(dataType), position++);
-                allParams.put(NameNormaliser.normaliseName(columnName), dbp);
+            dc.setString(1, objectName);
+            try (ResultSet rs = dc.executeQuery()) {
+                return createParamMap(rs);
             }
-            rs.close();
-            return allParams;
         }
     }
 
-    public Map<String, DbParameterAccessor> getAllProcedureParameters(
-            String procName) throws SQLException {
-        throw new UnsupportedOperationException();
+    private Map<String, DbParameterAccessor> createParamMap(ResultSet rs)
+            throws SQLException {
+        Map<String, DbParameterAccessor> allParams = new HashMap<>();
+        while (rs.next()) {
+            String paramName = defaultIfNull(rs.getString("COLUMN_NAME"), "");
+            String paramTypeName = rs.getString("TYPE_NAME");
+            allParams.put(NameNormaliser.normaliseName(paramName),
+                          createDbParameterAccessor(paramName,
+                                                    getParamDirection(rs.getString("COLUMN_DIRECTION")),
+                                                    typeMapper.getJDBCSQLTypeForDBType(paramTypeName),
+                                                    getJavaClass(paramTypeName),
+                                                    rs.getInt("ORDINAL_POSITION") - 1));
+        }
+        return allParams;
+    }
+
+    private Direction getParamDirection(String direction) throws SQLException {
+        switch (direction) {
+            case "INPUT":
+                return Direction.INPUT;
+            case "INPUT_OUTPUT":
+                return Direction.INPUT_OUTPUT;
+            case "OUTPUT":
+                return Direction.OUTPUT;
+            case "RETURN_VALUE":
+                return Direction.RETURN_VALUE;
+            default:
+                throw new SQLException("Invalid column/parameter direction string " + direction);
+        }
+    }
+
+    @Override
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(String procName)
+                                                                   throws SQLException {
+        // https://github.com/apache/derby/blob/10.13/java/engine/org/apache/derby/impl/jdbc/metadata.properties
+        String sql =
+        "SELECT CAST(NULL AS VARCHAR(128)) AS <<ProcFuncTypeName>>_CAT," +
+        "       S.SCHEMANAME               AS <<ProcFuncTypeName>>_SCHEM," +
```

Comment:
It makes sense the column aliases in the result to be the same for all the queries. (I'm not sure we need these specific columns for at all for what we're trying to do).